install/kubernetes/helm/subcharts/gateways/templates/deployment.yaml

```diff
@@ -143,7 +143,11 @@ spec:
           - name: ISTIO_META_POD_NAME
             valueFrom:
               fieldRef:
-                fieldPath: metadata.name             
+                fieldPath: metadata.name  
+          {{- if $.Values.global.proxy.network }}
+          - name: ISTIO_META_NETWORK
+            value: "{{ $.Values.global.proxy.network }}"
```

Comment:
??? where is this proxy.network? 
Also, why do you want to hardcode this? WE already have a facility to add environment variables (below). so why not add the environment variable to the remote gateway values.yaml

Reply:
This is defined in `proxy.network` in values.yaml (both istio and istio-remote) or provided with ` --set global.proxy.network`.
It's being used for the sidecar injector's config so added here too.
I can add it to env too but then users need to set in both places and frankly it's more a proxy var than gateway.

Reply:
I can move it from `proxy.network` to `multiCluster.network` but we already have `networks` for the configuration there so it will be a bit confusing.

Reply:
would not it be easier to define via helm config param via env var?

Reply:
Yeah.. thats what the other multicluster thing is doing. Seeing this might cause conflicts with the other multicluster solution

Reply:
> I can add it to env too but then users need to set in both places and frankly it's more a proxy var than gateway.
Also, that argument about aesthetics is really moot here given the convoluted setup involved for this shared control plane thing..

Reply:
Thinking more about this, I am pretty convinced that this should not be part of the gateway. Gateways are meant to be deployed in different environments, and used for variety of purposes. So, there is no point in inheriting a proxy's network on all gateways..

Reply:
@rshriram and any constructive suggestion?

Reply:
I agree we should not use env in values.yaml - it is meant as an escape for users to fine tune and enable experimental/special behaviors.

But we need a clean and consistent way to configure networks. This will end up in the sidecar injector and gateways. 

How about adding it directly under global: ? 

global:
   # Name of the network this deployment belongs to. 
   # Must match the names in the meshNetworks section.
   network: name